ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionProcessor.java

```diff
@@ -328,7 +334,14 @@ public TransactionProcessingResult processTransaction(
               transaction.getPayload(), transaction.isContractCreation());
       final long accessListGas =
           gasCalculator.accessListGasCost(accessListEntries.size(), accessListStorageCount);
-      final long gasAvailable = transaction.getGasLimit() - intrinsicGas - accessListGas;
+      final long dataGas;
+      if (feeMarket.implementsDataFee()) {
+        dataGas = gasCalculator.dataGasCost(transaction.getVersionedHashes().orElseThrow().size());
+      } else {
+        dataGas = 0L;
+      }
+
+      final long gasAvailable = transaction.getGasLimit() - intrinsicGas - accessListGas - dataGas;
       LOG.trace(
           "Gas available for execution {} = {} - {} (limit - intrinsic)",
```

Comment:
can we add dataGas to the log ?

Reply:
sure, will add also accessListGas that is missing too